eng/versioning.targets

```diff
@@ -76,6 +77,21 @@
     </ItemGroup>
   </Target>
 
+  <!-- Adds SupportedOSPlatform attribute to the assembly when:
+      * At least one <SupportedOSPlatform /> has been specified
+      * This isn't a test project
+      * This is a cross-platform target
+      * The build isn't targeting .NET Framework
+    -->
+  <Target Name="AddSupportedOSPlatformAttribute" BeforeTargets="GenerateAssemblyInfo" AfterTargets="PrepareForBuild"
```

Comment:
I wonder if we could do this in the above Target so we don't need to duplicate so much?

```xml
  <!-- Adds UnsupportedOSPlatform and SupportedOSPlatform attributes to the assembly when:
      * At least one <UnsupportedOSPlatforms /> or <SupportedOSPlatforms /> has been specified
      * This isn't a test project
      * This is a cross-platform target
      * The build isn't targeting .NET Framework
    -->
  <Target Name="AddOSPlatformAttributes" BeforeTargets="GenerateAssemblyInfo" AfterTargets="PrepareForBuild"
    Condition="'$(TargetsAnyOS)' == 'true' and '$(IsTestProject)' != 'true' and '$(TargetFrameworkIdentifier)' != '.NETFramework'">

    <ItemGroup Condition="'@(_unsupportedOSPlatforms)' != ''">
      <AssemblyAttribute Include="System.Runtime.Versioning.UnsupportedOSPlatform">
        <_Parameter1>%(_unsupportedOSPlatforms.Identity)</_Parameter1>
      </AssemblyAttribute>
    </ItemGroup>

    <ItemGroup Condition="'@(_supportedOSPlatforms)' != ''">
      <AssemblyAttribute Include="System.Runtime.Versioning.SupportedOSPlatform">
        <_Parameter1>%(_supportedOSPlatforms.Identity)</_Parameter1>
      </AssemblyAttribute>
    </ItemGroup>

  </Target>
```

Reply:
Absolutely, @buyaa-n would the proposed change be OK with you?

Reply:
Yeah, looks good